setup_environment.sh

```diff
@@ -0,0 +1,99 @@
+sudo tar -xvf jdk-7*
+sudo mv ./jdk1.7* /usr/lib/jvm/jdk1.7.0_79
+sudo update-alternatives --install "/usr/bin/java" "java" "/usr/lib/jvm/jdk1.7.0_79/bin/java" 1
+sudo update-alternatives --install "/usr/bin/javac" "javac" "/usr/lib/jvm/jdk1.7.0_79/bin/javac" 1
+sudo update-alternatives --install "/usr/bin/javaws" "javaws" "/usr/lib/jvm/jdk1.7.0_79/bin/javaws" 1
+sudo chmod a+x /usr/bin/java
+sudo chmod a+x /usr/bin/javac
+sudo chmod a+x /usr/bin/javaws
+
+
+# download, install and configure java8 jdk
+sudo wget --no-check-certificate --no-cookies --header "Cookie: oraclelicense=accept-securebackup-cookie" http://download.oracle.com/otn-pub/java/jdk/8u102-b14/jdk-8u102-linux-x64.tar.gz
+sudo tar -xvf jdk-8*
+sudo mv ./jdk1.8* /usr/lib/jvm/jdk1.8.0_102
+sudo update-alternatives --install "/usr/bin/java" "java" "/usr/lib/jvm/jdk1.8.0_102/bin/java" 1
+sudo update-alternatives --install "/usr/bin/javac" "javac" "/usr/lib/jvm/jdk1.8.0_102/bin/javac" 1
+sudo update-alternatives --install "/usr/bin/javaws" "javaws" "/usr/lib/jvm/jdk1.8.0_102/bin/javaws" 1
+sudo chmod a+x /usr/bin/java
+sudo chmod a+x /usr/bin/javac
+sudo chmod a+x /usr/bin/javaws
+
+printf "${RED}[IMPORTANT]${NOCOLOR} Pick JDK8 option!\n"
+sudo update-alternatives --config java
+printf "${RED}[IMPORTANT]${NOCOLOR} Pick JDK8 option!\n"
+sudo update-alternatives --config javac
+printf "${RED}[IMPORTANT]${NOCOLOR} Pick JDK8 option!\n"
+sudo update-alternatives --config javaws
+
+
+# set environmental variables
+sudo touch /etc/profile.d/oraclejdk.sh
+sudo chmod a+rwx /etc/profile.d/oraclejdk.sh
+sudo echo '#!/bin/bash' >> /etc/profile.d/oraclejdk.sh
+sudo echo "export PATH=\$PATH:/usr/lib/jvm/jdk1.8.0_102/bin" >> /etc/profile.d/oraclejdk.sh
+sudo echo "export JAVA_HOME=/usr/lib/jvm/jdk1.8.0_102" >> /etc/profile.d/oraclejdk.sh
+sudo echo "export JAVA7_HOME=/usr/lib/jvm/jdk1.7.0_79" >> /etc/profile.d/oraclejdk.sh
+sudo echo "export JAVA8_HOME=/usr/lib/jvm/jdk1.8.0_102" >> /etc/profile.d/oraclejdk.sh
+export PATH=$PATH:/usr/lib/jvm/jdk1.8.0_102/bin
+export JAVA_HOME=/usr/lib/jvm/jdk1.8.0_102
+export JAVA7_HOME=/usr/lib/jvm/jdk1.7.0_79
+export JAVA8_HOME=/usr/lib/jvm/jdk1.8.0_102
+
+# download, install and configure android sdk
+mkdir -p /home/$USER/android-sdk/
+
+sudo wget https://dl.google.com/android/android-sdk_r24.4.1-linux.tgz
+sudo tar -xvzf android-sdk_r24.4.1-linux.tgz
+
+cd android-sdk-linux/
+sudo mv * /home/$USER/android-sdk/
+
+sudo touch /etc/profile.d/android.sh
+sudo chmod a+rwx /etc/profile.d/android.sh
+
+sudo echo '#!/bin/bash' >> /etc/profile.d/android.sh
+sudo echo "export ANDROID_HOME=/home/${USER}/android-sdk/" >> /etc/profile.d/android.sh
+sudo echo "export PATH=\$PATH:/home/${USER}/android-sdk/tools" >> /etc/profile.d/android.sh
+export ANDROID_HOME=/home/$USER/android-sdk/
+export PATH=$PATH:/home/$USER/android-sdk/tools
+
+sudo chown -R $USER /home/$USER/android-sdk/
```

Comment:
You say:

> After unpacking android-sdk_r24* user has no rights to execute files in this directory, e.g. I couldn't run /tools/android later in script

If the file `android-sdk_r24*` is owned by the user and unpacked in the directory from which access rights are not inherited then I must say - you are wrong (see code snippet below).

There is also substantial difference between changing owner via `chown` (still I don't get why do you change ownership, who was the owner previously?) and changing access rights via `chmod` (but still after unpacking access rights are correct, see code below).

Note missing `v` in `tar` command.

```
# grzanka at gajewski-dell in ~/tmp [15:50:37]
→ wget https://dl.google.com/android/android-sdk_r24.4.1-linux.tgz
--2016-08-27 15:50:38--  https://dl.google.com/android/android-sdk_r24.4.1-linux.tgz
Resolving dl.google.com (dl.google.com)... 172.217.21.174, 2a00:1450:4001:81d::200e
Connecting to dl.google.com (dl.google.com)|172.217.21.174|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 326412652 (311M) [application/x-tar]
Saving to: ‘android-sdk_r24.4.1-linux.tgz’

100%[=========================================================================>] 326.412.652 5,38MB/s   in 62s    

2016-08-27 15:51:41 (5,02 MB/s) - ‘android-sdk_r24.4.1-linux.tgz’ saved [326412652/326412652]


# grzanka at gajewski-dell in ~/tmp [15:51:41]
→ tar -xzf android-sdk_r24.4.1-linux.tgz 

# grzanka at gajewski-dell in ~/tmp [15:52:02]
→ cd android-sdk-linux/

# grzanka at gajewski-dell in ~/tmp/android-sdk-linux [15:52:07]
→ mkdir -p $HOME/android-sdk

# grzanka at gajewski-dell in ~/tmp/android-sdk-linux [15:52:24]
→ mv * $HOME/android-sdk/

# grzanka at gajewski-dell in ~/tmp/android-sdk-linux [15:52:36]
→ cd $HOME/android-sdk/

# grzanka at gajewski-dell in ~/android-sdk [15:53:20]
→ ls -alh tools/android 
-rwxr--r-- 1 grzanka grzanka 3,5K paź 14  2015 tools/android

# grzanka at gajewski-dell in ~/android-sdk [15:53:31]
→ file tools/android
tools/android: POSIX shell script, ASCII text executable

# grzanka at gajewski-dell in ~/android-sdk [15:53:35]
→ ./tools/android --help

       Usage:
       android [global options] action [action options]
       Global options:
  -s --silent     : Silent mode, shows errors only.
  -v --verbose    : Verbose mode, shows errors, warnings and all messages.
     --clear-cache: Clear the SDK Manager repository manifest cache.
  -h --help       : Help on a specific command.

                                                                    Valid
                                                                    actions
                                                                    are
                                                                    composed
                                                                    of a verb
                                                                    and an
                                                                    optional
                                                                    direct
                                                                    object:
-    sdk              : Displays the SDK Manager window.
-    avd              : Displays the AVD Manager window.
-   list              : Lists existing targets or virtual devices.
-   list avd          : Lists existing Android Virtual Devices.
-   list target       : Lists existing targets.
-   list device       : Lists existing devices.
-   list sdk          : Lists remote SDK repository.
- create avd          : Creates a new Android Virtual Device.
-   move avd          : Moves or renames an Android Virtual Device.
- delete avd          : Deletes an Android Virtual Device.
- update avd          : Updates an Android Virtual Device to match the folders
                        of a new SDK.
- create project      : Creates a new Android project.
- update project      : Updates an Android project (must already have an
                        AndroidManifest.xml).
- create test-project : Creates a new Android project for a test package.
- update test-project : Updates the Android project for a test package (must
                        already have an AndroidManifest.xml).
- create lib-project  : Creates a new Android library project.
- update lib-project  : Updates an Android library project (must already have
                        an AndroidManifest.xml).
- create uitest-project: Creates a new UI test project.
- update adb          : Updates adb to support the USB devices declared in the
                        SDK add-ons.
- update sdk          : Updates the SDK by suggesting new platforms to install
                        if available.


```